### PR TITLE
Feature/1.5 allow custom sensibility for widget cache

### PR DESCRIPTION
**Pull request by Charlie-Lucas**

### CODE REVIEW

File: Bundle/WidgetBundle/Entity/Widget.php [hunk 1]
```diff
@@ -451,4 +451,16 @@ public function getView()
...
+
+    /**
+     * @return string
+     */
+    public function generateHash()
+    {
+        return sprintf('%s-%s-%s',
+            $this->getId(),
+            $this->getUpdatedAt()->getTimestamp(),
+            $this->getCurrentView()->getReference()->getId()
```
Comment (Charlie-Lucas) [inline Bundle/WidgetBundle/Entity/Widget.php]: Generate an exception

File: Bundle/BusinessPageBundle/Repository/BusinessPageRepository.php [hunk 2]
```diff
@@ -20,6 +20,9 @@ class BusinessPageRepository extends EntityRepository
      */
     public function findPageByBusinessEntityAndPattern(BusinessTemplate $pattern, $entity, BusinessEntity $businessEntity)
```
Comment (Charlie-Lucas) [inline Bundle/BusinessPageBundle/Repository/BusinessPageRepository.php]: php doc fill entity or entityID

File: Bundle/WidgetBundle/Entity/Widget.php [hunk 3]
```diff
@@ -451,4 +451,28 @@ public function getView()
     {
         return $this->view;
     }
+
+    /**
+     * @return string
+     */
+    public function generateHash()
```
Comment (lenybernard) [inline Bundle/WidgetBundle/Entity/Widget.php]: please rename into generateCacheId

**Comment by Charlie-Lucas** [comment]
separate different areas in several pull requests

**Pull request closed without merging**